sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubBufferedProducerAsyncClient.java

```diff
@@ -124,7 +124,7 @@ public final class EventHubBufferedProducerAsyncClient implements Closeable {
             return new ArrayList<>(partitionProducers.keySet()).toArray(new String[0]);
         })).cache();
 
-        this.tracer = tracer;
+        this.tracer = builder.createTracer();
```

Comment:
I'm curious why we removed the ctor parameter for Tracer? I feel like it is a dependency of the BufferedProducer and is easier to mock than having to mock the builder and then its subsequent return methods.